src/test/java/com/example/springboojpa/CustomerPersistenceTest.java

```diff
@@ -0,0 +1,96 @@
+package com.example.springboojpa;
+
+import com.example.springboojpa.domain.customer.Customer;
+import jakarta.persistence.EntityManager;
+import jakarta.persistence.EntityManagerFactory;
+import jakarta.persistence.EntityTransaction;
+import jakarta.transaction.Transactional;
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+
+@SpringBootTest
+@Transactional
```

Comment:
```suggestion
@DataJpaTest
@AutoConfigureTestDatabase(replace = AutoConfigureTestDatabase.Replace.NONE)
```

@SpringBootTest + @Transactional은 @DataJpaTest로 대체 가능하고,
EmbeddedDataBase를 사용하지 않고, MySQL 개발환경을 별도로 구축하였기 때문에

@AutoConfigureTestDatabase(replace = AutoConfigureTestDatabase.Replace.NONE)
를 사용하면 될 것 같네요.!

다만 전체 테스트를 실행해보면 save, find가 실패하는 것으로 보여집니다.

테스트 순서가 보장되어야 한다면 
@TestInstance(TestInstance.Lifecycle.PER_CLASS)
@TestMethodOrder(value = MethodOrderer.OrderAnnotation.class)
를 선언하여

각 테스트에 @Order(N)으로 순서를 보장하는 방법으로 구현하는 것이 좋을 것 같습니다.